modules/apps/blogs/blogs-service/src/main/java/com/liferay/blogs/internal/model/listener/UserGroupModelListener.java

```diff
@@ -58,14 +58,15 @@ private void _unsubscribeUserFromUserGroupGroups(
 			long userId, long userGroupId)
 		throws PortalException {
 
-		List<Group> userGroupGroups = _groupLocalService.getUserGroupGroups(
+		long[] userGroupGroupIds = _userGroupLocalService.getGroupPrimaryKeys(
 			userGroupId);
 
-		userGroupGroups.removeAll(
-			_groupLocalService.getUserGroups(userId, true));
+		List<Group> userGroups = _groupLocalService.getUserGroups(userId, true);
 
-		for (Group group : userGroupGroups) {
-			_blogsEntryLocalService.unsubscribe(userId, group.getGroupId());
+		for (long groupId : userGroupGroupIds) {
+			if (!userGroups.contains(groupId)) {
+				_blogsEntryLocalService.unsubscribe(userId, groupId);
+			}
 		}
 	}
```

Comment:
@marianoalvarosaiz, `userGroups.contains(groupId)` is different object. For my previous provided suggestion which is not enough, sorry for this. The following codes should be right.
```
List<Long> groupIds = ListUtil.toList(
			_groupLocalService.getUserGroups(userId, true), Group::getGroupId);

		for (long groupId :
				_userGroupLocalService.getGroupPrimaryKeys(userGroupId)) {

			if (!groupIds.contains(Long.valueOf(groupId))) {
				_blogsEntryLocalService.unsubscribe(userId, groupId);
			}
		}
```